apps/web/src/components/NavMenu/useMainNavLinks.tsx

```diff
@@ -0,0 +1,330 @@
+      subMenu
+    />
+  );
+
+  const CareerTimeline = (
+    <NavItem
+      key="careerTimeline"
+      href={paths.careerTimelineAndRecruitment()}
+      title={intl.formatMessage(
+        navigationMessages.careerTimelineAndRecruitment,
+      )}
+      subMenu
+    />
+  );
+
+  const SkillLibrary = (
+    <NavItem
+      key="skillLibrary"
+      href={paths.skillPortfolio()}
+      title={intl.formatMessage(navigationMessages.skillPortfolio)}
+      subMenu
+    />
+  );
+
+  const ManagerProfile = (
+    <NavItem
+      key="managerProfile"
+      href={paths.profile()}
+      title={intl.formatMessage(navigationMessages.managerProfile)}
+      subMenu
+    />
+  );
+
+  const AccountSettings = (
+    <NavItem
+      key="accountSettings"
+      href={paths.accountSettings()}
+      title={intl.formatMessage(navigationMessages.accountSettings)}
+      subMenu
+    />
+  );
+
+  const SignIn = (
+    <NavItem
+      key="signIn"
+      href={paths.login()}
+      title={intl.formatMessage(authMessages.signIn)}
+    />
+  );
+
+  const SignUp = (
+    <NavItem
+      key="signUp"
+      href={paths.register()}
+      title={intl.formatMessage(authMessages.signUp)}
+    />
+  );
+
+  const SignOut = (
+    <SignOutConfirmation key="sign-out">
+      <LogoutButton>{intl.formatMessage(authMessages.signOut)}</LogoutButton>
+    </SignOutConfirmation>
+  );
+
+  const getRoleName: Record<string, string> = {
+    ["applicant"]: intl.formatMessage(navMenuMessages.applicant),
+    ["manager"]: intl.formatMessage(navMenuMessages.manager),
+    ["pool_operator"]: intl.formatMessage(navMenuMessages.community),
+    ["request_responder"]: intl.formatMessage(navMenuMessages.community),
+    ["community_manager"]: intl.formatMessage(navMenuMessages.community),
+    ["process_operator"]: intl.formatMessage(navMenuMessages.community),
+    ["community_recruiter"]: intl.formatMessage(navMenuMessages.community),
+    ["community_admin"]: intl.formatMessage(navMenuMessages.community),
+    ["platform_admin"]: intl.formatMessage(navMenuMessages.admin),
+    [""]: intl.formatMessage(commonMessages.notFound),
+  };
+
+  const roles: Record<string, string> = {
+    ["guest"]: paths.home(),
+    ["applicant"]: paths.applicantDashboard(),
+    ["manager"]: paths.manager(),
+    ["pool_operator"]: paths.communityDashboard(),
+    ["request_responder"]: paths.communityDashboard(),
+    ["community_manager"]: paths.communityDashboard(),
+    ["process_operator"]: paths.communityDashboard(),
+    ["community_recruiter"]: paths.communityDashboard(),
+    ["community_admin"]: paths.communityDashboard(),
+    ["platform_admin"]: paths.adminDashboard(),
+    [""]: paths.notFound(),
+  };
+
+  const roleLinks = roleAssignments
+    .filter(
+      (roleAssignment) => roleAssignment.role?.name !== ROLE_NAME.BaseUser,
+    )
+    .map((roleAssignment) => {
+      const role = roleAssignment.role?.name ?? "";
```

Comment:
I'd recommend just testing for falsy here and return early instead of putting junk rows in your map objects.

Reply:
Yea that's better :+1: [1ea6670](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/1ea6670ccf30682052a5c179128a2355c047bffe)